system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -613,52 +619,58 @@ bool ElasticStackLogAccess::fetchLog(const LogAccessConditions & options, String
 class ELASTICSTACKLOGACCESS_API ElasticStackLogStream : public CInterfaceOf<IRemoteLogAccessStream>
 {
 public:
-    virtual bool readLogEntries(StringBuffer & record, unsigned & recsRead, unsigned & totalRecs) override
+    virtual unsigned logQueryTotalCount() override
+    {
+        UNIMPLEMENTED;
+        //return m_totalRecs;
+    }
+
+    virtual bool readLogEntries(StringBuffer & record, unsigned & recsRead) override
     {
         Json::Value res;
         recsRead = 0;
-        totalRecs = 0;
 
         if (m_esSroller.next(res))
         {
-
-            if (!res["total"]["value"].empty())
-            {
-                totalRecs = res["total"]["value"].asUInt();
-            }
+            //if (!res["total"]["value"].empty())
+            //{
+            //    m_totalRecs = res["total"]["value"].asUInt();
+            //}
 
             if (!res["hits"].empty())
             {
                 recsRead = res["hits"].size();
                 std::ostringstream sout;
                 m_jsonWriter->write(res, &sout);
-                processESScrollJsonResp(sout.str().c_str(), record, m_outputFormat);
+                processESScrollJsonResp(sout.str().c_str(), record, m_outputFormat, false);
                 return true;
             }
         }
 
         return false;
     }
 
-    ElasticStackLogStream(std::string & queryString, const char * connstr, const char * indexsearchpattern, LogAccessLogFormat format,  std::size_t pageSize, std::string scrollTO)
-     : m_esSroller(std::make_shared<elasticlient::Client>(std::vector<std::string>({connstr})), pageSize, scrollTO)
+    ElasticStackLogStream(std::string & queryString, const char * connstr, const char * indexsearchpattern, LogAccessLogFormat format,  std::size_t pageSize, std::string scrollTo)
+     : m_esSroller(std::make_shared<elasticlient::Client>(std::vector<std::string>({connstr})), pageSize, scrollTo)
     {
         m_outputFormat = format;
         m_esSroller.init(indexsearchpattern, DEFAULT_ES_DOC_TYPE, queryString);
-        m_jsonWriter = m_jsonStreamBuilder.newStreamWriter();
+        m_jsonWriter = std::unique_ptr<Json::StreamWriter>(m_jsonStreamBuilder.newStreamWriter());
```

Comment:
minor: this can be more concisely expressed as:

jsonWriter.reset(m_jsonStreamBuilder.newStreamWriter());